packages/opencensus-nodejs/src/trace/instrumentation/plugin-loader.ts

```diff
@@ -102,30 +113,37 @@ export class PluginLoader {
    * @param pluginList A list of plugins.
    */
   loadPlugins(pluginList: PluginNames) {
-    // tslint:disable:no-any
-    hook(Object.keys(pluginList), (exports, name, basedir) => {
-      const version = this.getPackageVersion(name, basedir as string);
-      this.logger.info('trying loading %s.%s', name, version);
-      let moduleExports = exports;
-      if (!version) {
-        return moduleExports;
-      } else {
-        this.logger.debug('applying patch to %s@%s module', name, version);
-        this.logger.debug(
-            'using package %s to patch %s', pluginList[name], name);
-        // Expecting a plugin from module;
-        try {
-          const plugin: Plugin = require(pluginList[name]).plugin;
-          this.plugins.push(plugin);
-          moduleExports = plugin.enable(exports, this.tracer, version, basedir);
-        } catch (e) {
-          this.logger.error(
-              'could not load plugin %s of module %s. Error: %s',
-              pluginList[name], name, e.message);
+    if (this.hookState === HookState.UNINITIALIZED) {
+      hook(Object.keys(pluginList), (exports, name, basedir) => {
+        if (this.hookState === HookState.ENABLED) {
+          const version = this.getPackageVersion(name, basedir as string);
+          this.logger.info('trying loading %s.%s', name, version);
+          let moduleExports = exports;
+          if (!version) {
+            return moduleExports;
+          } else {
+            this.logger.debug('applying patch to %s@%s module', name, version);
+            this.logger.debug(
+                'using package %s to patch %s', pluginList[name], name);
+            // Expecting a plugin from module;
+            try {
+              const plugin: Plugin = require(pluginList[name]).plugin;
+              this.plugins.push(plugin);
+              moduleExports =
+                  plugin.enable(exports, this.tracer, version, basedir);
+            } catch (e) {
+              this.logger.error(
+                  'could not load plugin %s of module %s. Error: %s',
+                  pluginList[name], name, e.message);
+            }
+            return moduleExports;
+          }
+        } else {
+          return exports;
         }
-        return moduleExports;
-      }
-    });
+      });
+      this.hookState = HookState.ENABLED;
```

Comment:
Make this the first line of the if uninitialize block; otherwise early returns miss this.

Reply:
Actually the hook callback is executed on a different tick. But that being said, thanks for highlighting this as I should be bringing this out of the `if` block entirely (otherwise re-activation won't work).